// gen/envoy/service/metrics/v3/metrics_service_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var envoy_service_metrics_v3_metrics_service_pb = require('../../../../envoy/service/metrics/v3/metrics_service_pb.js');
var envoy_config_core_v3_base_pb = require('../../../../envoy/config/core/v3/base_pb.js');
var io_prometheus_client_metrics_pb = require('../../../../io/prometheus/client/metrics_pb.js');
var udpa_annotations_status_pb = require('../../../../udpa/annotations/status_pb.js');
var udpa_annotations_versioning_pb = require('../../../../udpa/annotations/versioning_pb.js');
var validate_validate_pb = require('../../../../validate/validate_pb.js');

function serialize_envoy_service_metrics_v3_StreamMetricsMessage(arg) {
  if (!(arg instanceof envoy_service_metrics_v3_metrics_service_pb.StreamMetricsMessage)) {
    throw new Error('Expected argument of type envoy.service.metrics.v3.StreamMetricsMessage');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_metrics_v3_StreamMetricsMessage(buffer_arg) {
  return envoy_service_metrics_v3_metrics_service_pb.StreamMetricsMessage.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_metrics_v3_StreamMetricsResponse(arg) {
  if (!(arg instanceof envoy_service_metrics_v3_metrics_service_pb.StreamMetricsResponse)) {
    throw new Error('Expected argument of type envoy.service.metrics.v3.StreamMetricsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_metrics_v3_StreamMetricsResponse(buffer_arg) {
  return envoy_service_metrics_v3_metrics_service_pb.StreamMetricsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// [#protodoc-title: Metrics service]
//
// Service for streaming metrics to server that consumes the metrics data. It uses Prometheus metric
// data model as a standard to represent metrics information.
var MetricsServiceService = exports.MetricsServiceService = {
  // Envoy will connect and send StreamMetricsMessage messages forever. It does not expect any
// response to be sent as nothing would be done in the case of failure.
streamMetrics: {
    path: '/envoy.service.metrics.v3.MetricsService/StreamMetrics',
    requestStream: true,
    responseStream: false,
    requestType: envoy_service_metrics_v3_metrics_service_pb.StreamMetricsMessage,
    responseType: envoy_service_metrics_v3_metrics_service_pb.StreamMetricsResponse,
    requestSerialize: serialize_envoy_service_metrics_v3_StreamMetricsMessage,
    requestDeserialize: deserialize_envoy_service_metrics_v3_StreamMetricsMessage,
    responseSerialize: serialize_envoy_service_metrics_v3_StreamMetricsResponse,
    responseDeserialize: deserialize_envoy_service_metrics_v3_StreamMetricsResponse,
  },
};

exports.MetricsServiceClient = grpc.makeGenericClientConstructor(MetricsServiceService);
