// gen/envoy/service/discovery/v3/ads_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var envoy_service_discovery_v3_ads_pb = require('../../../../envoy/service/discovery/v3/ads_pb.js');
var envoy_service_discovery_v3_discovery_pb = require('../../../../envoy/service/discovery/v3/discovery_pb.js');
var udpa_annotations_status_pb = require('../../../../udpa/annotations/status_pb.js');
var udpa_annotations_versioning_pb = require('../../../../udpa/annotations/versioning_pb.js');

function serialize_envoy_service_discovery_v3_DeltaDiscoveryRequest(arg) {
  if (!(arg instanceof envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryRequest)) {
    throw new Error('Expected argument of type envoy.service.discovery.v3.DeltaDiscoveryRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_discovery_v3_DeltaDiscoveryRequest(buffer_arg) {
  return envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_discovery_v3_DeltaDiscoveryResponse(arg) {
  if (!(arg instanceof envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryResponse)) {
    throw new Error('Expected argument of type envoy.service.discovery.v3.DeltaDiscoveryResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_discovery_v3_DeltaDiscoveryResponse(buffer_arg) {
  return envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_discovery_v3_DiscoveryRequest(arg) {
  if (!(arg instanceof envoy_service_discovery_v3_discovery_pb.DiscoveryRequest)) {
    throw new Error('Expected argument of type envoy.service.discovery.v3.DiscoveryRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_discovery_v3_DiscoveryRequest(buffer_arg) {
  return envoy_service_discovery_v3_discovery_pb.DiscoveryRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_discovery_v3_DiscoveryResponse(arg) {
  if (!(arg instanceof envoy_service_discovery_v3_discovery_pb.DiscoveryResponse)) {
    throw new Error('Expected argument of type envoy.service.discovery.v3.DiscoveryResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_discovery_v3_DiscoveryResponse(buffer_arg) {
  return envoy_service_discovery_v3_discovery_pb.DiscoveryResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// [#protodoc-title: Aggregated Discovery Service (ADS)]
//
// Discovery services for endpoints, clusters, routes,
// and listeners are retained in the package `envoy.api.v2` for backwards
// compatibility with existing management servers. New development in discovery
// services should proceed in the package `envoy.service.discovery.v2`.
//
// See https://github.com/envoyproxy/envoy-api#apis for a description of the role of
// ADS and how it is intended to be used by a management server. ADS requests
// have the same structure as their singleton xDS counterparts, but can
// multiplex many resource types on a single stream. The type_url in the
// DiscoveryRequest/DiscoveryResponse provides sufficient information to recover
// the multiplexed singleton APIs at the Envoy instance and management server.
var AggregatedDiscoveryServiceService = exports.AggregatedDiscoveryServiceService = {
  // This is a gRPC-only API.
streamAggregatedResources: {
    path: '/envoy.service.discovery.v3.AggregatedDiscoveryService/StreamAggregatedResources',
    requestStream: true,
    responseStream: true,
    requestType: envoy_service_discovery_v3_discovery_pb.DiscoveryRequest,
    responseType: envoy_service_discovery_v3_discovery_pb.DiscoveryResponse,
    requestSerialize: serialize_envoy_service_discovery_v3_DiscoveryRequest,
    requestDeserialize: deserialize_envoy_service_discovery_v3_DiscoveryRequest,
    responseSerialize: serialize_envoy_service_discovery_v3_DiscoveryResponse,
    responseDeserialize: deserialize_envoy_service_discovery_v3_DiscoveryResponse,
  },
  deltaAggregatedResources: {
    path: '/envoy.service.discovery.v3.AggregatedDiscoveryService/DeltaAggregatedResources',
    requestStream: true,
    responseStream: true,
    requestType: envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryRequest,
    responseType: envoy_service_discovery_v3_discovery_pb.DeltaDiscoveryResponse,
    requestSerialize: serialize_envoy_service_discovery_v3_DeltaDiscoveryRequest,
    requestDeserialize: deserialize_envoy_service_discovery_v3_DeltaDiscoveryRequest,
    responseSerialize: serialize_envoy_service_discovery_v3_DeltaDiscoveryResponse,
    responseDeserialize: deserialize_envoy_service_discovery_v3_DeltaDiscoveryResponse,
  },
};

exports.AggregatedDiscoveryServiceClient = grpc.makeGenericClientConstructor(AggregatedDiscoveryServiceService);
