// gen/envoy/service/ratelimit/v3/rls_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var envoy_service_ratelimit_v3_rls_pb = require('../../../../envoy/service/ratelimit/v3/rls_pb.js');
var envoy_config_core_v3_base_pb = require('../../../../envoy/config/core/v3/base_pb.js');
var envoy_extensions_common_ratelimit_v3_ratelimit_pb = require('../../../../envoy/extensions/common/ratelimit/v3/ratelimit_pb.js');
var google_protobuf_duration_pb = require('google-protobuf/google/protobuf/duration_pb.js');
var google_protobuf_struct_pb = require('google-protobuf/google/protobuf/struct_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');
var udpa_annotations_status_pb = require('../../../../udpa/annotations/status_pb.js');
var udpa_annotations_versioning_pb = require('../../../../udpa/annotations/versioning_pb.js');
var validate_validate_pb = require('../../../../validate/validate_pb.js');

function serialize_envoy_service_ratelimit_v3_RateLimitRequest(arg) {
  if (!(arg instanceof envoy_service_ratelimit_v3_rls_pb.RateLimitRequest)) {
    throw new Error('Expected argument of type envoy.service.ratelimit.v3.RateLimitRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_ratelimit_v3_RateLimitRequest(buffer_arg) {
  return envoy_service_ratelimit_v3_rls_pb.RateLimitRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_ratelimit_v3_RateLimitResponse(arg) {
  if (!(arg instanceof envoy_service_ratelimit_v3_rls_pb.RateLimitResponse)) {
    throw new Error('Expected argument of type envoy.service.ratelimit.v3.RateLimitResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_ratelimit_v3_RateLimitResponse(buffer_arg) {
  return envoy_service_ratelimit_v3_rls_pb.RateLimitResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// [#protodoc-title: Rate limit service (RLS)]
//
var RateLimitServiceService = exports.RateLimitServiceService = {
  // Determine whether rate limiting should take place.
shouldRateLimit: {
    path: '/envoy.service.ratelimit.v3.RateLimitService/ShouldRateLimit',
    requestStream: false,
    responseStream: false,
    requestType: envoy_service_ratelimit_v3_rls_pb.RateLimitRequest,
    responseType: envoy_service_ratelimit_v3_rls_pb.RateLimitResponse,
    requestSerialize: serialize_envoy_service_ratelimit_v3_RateLimitRequest,
    requestDeserialize: deserialize_envoy_service_ratelimit_v3_RateLimitRequest,
    responseSerialize: serialize_envoy_service_ratelimit_v3_RateLimitResponse,
    responseDeserialize: deserialize_envoy_service_ratelimit_v3_RateLimitResponse,
  },
};

exports.RateLimitServiceClient = grpc.makeGenericClientConstructor(RateLimitServiceService);
