// gen/envoy/service/accesslog/v3/als_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var envoy_service_accesslog_v3_als_pb = require('../../../../envoy/service/accesslog/v3/als_pb.js');
var envoy_config_core_v3_base_pb = require('../../../../envoy/config/core/v3/base_pb.js');
var envoy_data_accesslog_v3_accesslog_pb = require('../../../../envoy/data/accesslog/v3/accesslog_pb.js');
var udpa_annotations_status_pb = require('../../../../udpa/annotations/status_pb.js');
var udpa_annotations_versioning_pb = require('../../../../udpa/annotations/versioning_pb.js');
var validate_validate_pb = require('../../../../validate/validate_pb.js');

function serialize_envoy_service_accesslog_v3_StreamAccessLogsMessage(arg) {
  if (!(arg instanceof envoy_service_accesslog_v3_als_pb.StreamAccessLogsMessage)) {
    throw new Error('Expected argument of type envoy.service.accesslog.v3.StreamAccessLogsMessage');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_accesslog_v3_StreamAccessLogsMessage(buffer_arg) {
  return envoy_service_accesslog_v3_als_pb.StreamAccessLogsMessage.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_envoy_service_accesslog_v3_StreamAccessLogsResponse(arg) {
  if (!(arg instanceof envoy_service_accesslog_v3_als_pb.StreamAccessLogsResponse)) {
    throw new Error('Expected argument of type envoy.service.accesslog.v3.StreamAccessLogsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_envoy_service_accesslog_v3_StreamAccessLogsResponse(buffer_arg) {
  return envoy_service_accesslog_v3_als_pb.StreamAccessLogsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// [#protodoc-title: gRPC access log service (ALS)]
//
// Service for streaming access logs from Envoy to an access log server.
var AccessLogServiceService = exports.AccessLogServiceService = {
  // Envoy will connect and send StreamAccessLogsMessage messages forever. It does not expect any
// response to be sent as nothing would be done in the case of failure. The server should
// disconnect if it expects Envoy to reconnect. In the future we may decide to add a different
// API for "critical" access logs in which Envoy will buffer access logs for some period of time
// until it gets an ACK so it could then retry. This API is designed for high throughput with the
// expectation that it might be lossy.
streamAccessLogs: {
    path: '/envoy.service.accesslog.v3.AccessLogService/StreamAccessLogs',
    requestStream: true,
    responseStream: false,
    requestType: envoy_service_accesslog_v3_als_pb.StreamAccessLogsMessage,
    responseType: envoy_service_accesslog_v3_als_pb.StreamAccessLogsResponse,
    requestSerialize: serialize_envoy_service_accesslog_v3_StreamAccessLogsMessage,
    requestDeserialize: deserialize_envoy_service_accesslog_v3_StreamAccessLogsMessage,
    responseSerialize: serialize_envoy_service_accesslog_v3_StreamAccessLogsResponse,
    responseDeserialize: deserialize_envoy_service_accesslog_v3_StreamAccessLogsResponse,
  },
};

exports.AccessLogServiceClient = grpc.makeGenericClientConstructor(AccessLogServiceService);
